### DISCUSSION TITLE
fprime-community/system-reference Building libcamera component

### DISCUSSION THREAD

**Discussion by lukeclements** [nasa/fprime#2359] [Q&A] [answered]
Hi, I am using the fprime-system-refernece libcamera component and when building I get this error:

```
/opt/cross-pi-gcc-10.3.0-0/bin/../lib/gcc/arm-linux-gnueabihf/10.3.1/../../../../arm-linux-gnueabihf/bin/ld: warning: libatomic.so.1, needed by ../../lib/raspberrypi/libcamera/build/lib/libcamera.so, not found (try using -rpath or -rpath-link)
/opt/cross-pi-gcc-10.3.0-0/bin/../lib/gcc/arm-linux-gnueabihf/10.3.1/../../../../arm-linux-gnueabihf/bin/ld: ../../lib/raspberrypi/libcamera/build/lib/libcamera.so: undefined reference to `__atomic_fetch_add_8@LIBATOMIC_1.0'
collect2: error: ld returned 1 exit status
gmake[2]: *** [RpiFSW/CMakeFiles/RpiFSW.dir/build.make:155: bin/raspberrypi/RpiFSW] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:8335: RpiFSW/CMakeFiles/RpiFSW.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
[ERROR] CMake erred with return code 2
```

The undefined reference to latomic error, indicates that the binary executable isn't properly linked to the "latomic" package provided by the cross-compiler (which I've confirmed it has). To resolve the error I have to put a "-latomic" flag and it successfully builds, but since I am running that command manually, it's not automatically copying into build artifacts. So I have to get the binary from the build-fprime-automatic directory as well as the XML instead of the build-artifacts directory.

It works, but is there a way to add the latomic flag in the CMake files so I don't have to do it manually?

**Comment by LeStarch** [comment, marked as answer]
Yes, you can add: `-latomic` to your `MOD_DEPS` list for your executable (wherever you call register_fprime_deployment).

**Reply by lukeclements** [reply]
That worked, thank you!
